apidef/oas/middleware.go

```diff
@@ -720,3 +737,34 @@ func (et *EnforceTimeout) ExtractTo(meta *apidef.HardTimeoutMeta) {
 	meta.Disabled = !et.Enabled
 	meta.TimeOut = et.Value
 }
+
+type CustomPluginMiddleware struct {
+	// FunctionName is the name of authentication method.
+	FunctionName string `bson:"functionName,omitempty" json:"functionName,omitempty"`
```

Comment:
Is there a default value for `functionName` and `path`? If not, why `omitempty`? Shouldn't they be required?

Reply:
yes, good catch. 
the function name and path - both are required.